File: week2_algorithmic_warmup/4_least_common_multiple.py
# Uses python2
import sys
import random

def greatest_common_divisor_fast(a, b):
    if b == 0:
        return a
    divisor = a % b
    return greatest_common_divisor_fast(b, divisor)

def least_common_multiple_fast(a, b):
    gcd = greatest_common_divisor_fast(a, b)
    return long((a*b)/gcd)

if __name__ == '__main__':
    input = sys.stdin.read();
    a, b = map(int, input.split())
    print(least_common_multiple_fast(a, b))

File: week2_algorithmic_warmup/4_least_common_multiple.js
const greatest_common_divisor_fast = function(a,b) {
  let aa = Math.max(a, b)
  let bb = Math.min(a, b)
  let divisor = aa % bb
  if(bb === 0) return aa
  return greatest_common_divisor_fast(bb,divisor)
}

data = '';

process.stdin.on('readable', function() {
  var chunk;
  while (chunk = process.stdin.read()) {
    data += chunk;
  }
});

process.stdin.on('end', function () {

  let lines = data.split(/\r\n|\r|\n|\\n/)
  let input = lines[0].split(' ')

  const least_common_multiple_fast = function(a,b) {
    const gcd = greatest_common_divisor_fast(a,b);
    return (a*b)/gcd
  }

  const greatest_common_divisor_fast = function(a,b) {
    let aa = Math.max(a, b)
    let bb = Math.min(a, b)
    let divisor = aa % bb
    if(bb === 0) return aa
    return greatest_common_divisor_fast(bb,divisor)
  }

  console.log(least_common_multiple_fast(input[0], input[1]))
});
